pkg/controller/node/nodecontroller.go

```diff
@@ -303,13 +303,14 @@ func (nc *NodeController) monitorNodeStatus() error {
 				if _, err := instances.ExternalID(node.Name); err != nil && err == cloudprovider.InstanceNotFound {
 					glog.Infof("Deleting node (no longer present in cloud provider): %s", node.Name)
 					nc.recordNodeEvent(node.Name, fmt.Sprintf("Deleting Node %v because it's not present according to cloud provider", node.Name))
+					if err := nc.deletePods(node.Name); err != nil {
```

Comment:
It might be worth a comment here that the node isn't deleted until all pods assigned to the node have been deleted.